fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -258,14 +258,9 @@ public CompletableFuture<RecordCursorResult<IndexEntry>> onNext() {
     @Nonnull
     private RecordCursorContinuation continuationHelper() {
         if (currentPosition >= topDocs.scoreDocs.length && limitRemaining > 0) {
-            return ByteArrayContinuation.fromNullable(null);
+            return RecordCursorEndContinuation.END;
         } else {
-            LuceneContinuationProto.LuceneIndexContinuation continuation = LuceneContinuationProto.LuceneIndexContinuation.newBuilder()
-                    .setDoc(searchAfter.doc)
-                    .setScore(searchAfter.score)
-                    .setShard(searchAfter.shardIndex)
-                    .build();
-            return ByteArrayContinuation.fromNullable(continuation.toByteArray());
+            return LuceneCursorContinuation.fromScoreDoc(searchAfter);
```

Comment:
I abstracted this kind of continuation into its own class. This change preserves the continuation value, but it means we will no longer need to serialize the proto to a byte array (if the continuation isn't needed)